tests/Feature/EditPageTest.php

```diff
@@ -0,0 +1,191 @@
+
+        // Find the edit page
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+        ;
+        // See the names in the page
+        foreach ($carers as $sec_carer) {
+             $this->see($sec_carer->name)
+                 ->seeElement('input[type="hidden"][value="'. $sec_carer->name .'"]')
+                 ;
+        }
+    }
+
+
+    /** @test */
+    public function itShowsAChildInputComplex()
+    {
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeElement('input[name="dob-month"]')
+            ->seeElement('input[name="dob-year"]')
+            ->seeElement('button[id="add-dob"]')
+        ;
+    }
+
+    /** @test */
+    public function itShowsAListOfChildren()
+    {
+        // Clear the children
+        $this->registration->family->children()->delete();
+        // Make 4 more
+        $new_children = factory(App\Child::class, 4)->make();
+        // Add to Family
+        $this->registration->family->children()->saveMany($new_children);
+
+        // Get the children again
+        $children = $this->registration->family->children;
+
+        // There should be 4...
+        $this->assertTrue($children->count() == 4);
+
+        // Find the edit page
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+        ;
+        // See the names in the page
+        foreach ($children as $child) {
+            $this->see('<td>'. $child->getAgeString() .'</td>')
+                ->see('<td>'. $child->getDobAsString() .'</td>')
+                ->see('<td>'. $child->getStatusString() .'</td>')
+                ->seeElement('input[type="hidden"][value="'. $child->dob->format('Y-m') .'"]')
+            ;
+        }
+    }
+
+    /** @test */
+    public function itShowsAFormSaveButton()
+    {
+        $this->actingAs($this->user)
+            ->visit(URL::route('service.registration.edit', [ 'id' => $this->registration ]))
+            ->seeInElement('button[type=submit]', 'Save Changes')
+        ;
+    }
+
+    /** @test */
+    public function itOnlyShowsAFoodMattersInputsToFoodMattersUsers()
+    {
+        // Create a FM User
+        $users[] = factory(App\User::class)->create([
+            "name"  => "test FM user",
+            "email" => "testufmser@example.com",
+            "password" => bcrypt('test_fm_user_pass'),
+            "centre_id" => $this->centre->id,
+            "role" => 'foodmatters_user',
+        ]);
+
+        $users[] = factory(App\User::class)->create([
+            "name"  => "test cc user",
+            "email" => "testccuser@example.com",
+            "password" => bcrypt('test_cc_user_pass'),
+            "centre_id" => $this->centre->id,
+            "role" => 'centre_user',
+        ]);
+
+        foreach ($users as $user) {
```

Comment:
Would it be better or additionally good to make your 2 types of user and then expect them to see what they can? Rather than as this does testing that the guard is working?